src/lib/core/fiber.h

```diff
@@ -281,6 +281,27 @@ fiber_yield(void);
 API_EXPORT void
 fiber_start(struct fiber *callee, ...);
 
+/**
+ * Set an argument for the fiber function. Can be used to avoid calling
+ * fiber_start which means no yields.
+ *
+ * \param f     fiber to set the argument for
+ * \param f_arg argument for the fiber function
+ */
+API_EXPORT void
+fiber_set_arg(struct fiber *f, void *f_arg);
+
+/**
+ * Get the argument for the current fiber which was set via the fiber_set_arg
+ * function. Can be used to avoid calling fiber_start which means no yields.
+ *
+ * \retval      argument for the fiber function set by fiber_set_arg function
+ *
+ * \sa fiber_set_arg
+ */
+API_EXPORT void *
+fiber_get_arg();
```

Comment:
In C we use `void` in place of arguments when a function doesn't have any.

By the way, it might also be possible to make this function take a fiber pointer. It would allow to use it not only on self-fiber. I only hesitate because most of the other fiber functions also don't take a fiber pointer unless it is the only way (for example, `fiber_yield()` makes no sense on non-self fiber, hence it doesn't take a fiber pointer argument). I am fine with both your solution and if you decide to add the fiber pointer argument.

Reply:
added the fiber pointer argument